### PR TITLE
Extract libraries from ci-style-fixer

### PR DESCRIPTION
**Pull request by Chekote**
For Medology/Umami#88

## Problem
There are functions within the ci-style-fixer that can be used by other scripts either in submodules or the host projects that need to work with Github, CircleCI, Git etc. Since these functions are within the ci-style-fixer script, it is not possible to utilize them from other scripts.

## Fix
Extract the functions into bash library scripts within the `lib` folder so that they can be re-used.

### CODE REVIEW

File: lib/github.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        -S "https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER"
+}
+
+#######################################
+# Extracts the URL from a pull request details response
+#
+# Arguments:
+#   1 the pull request details response to extract the URL from
+#######################################
+github::pull::url::get () {
```
Comment (manuerumx) [inline lib/github.sh:40]: Can you explain to me why is defined with colons, but is invoked using only a dot?

Definition: `github::pull::url::get`
Invocation: `git_url=$(github.pull.url.get "$api_response")`

To be consistent shouldn't be declared with dots like in the git.sh file?

**Review by manuerumx — CHANGES_REQUESTED** [review]
Just a question about the implementation

File: lib/github.sh (hunk 1 above)
Comment (Chekote) [reply lib/github.sh:40]: Yes, I missed this one when I changed from :: to . in the final commit. I will fix it.

**Review by manuerumx — APPROVED** [review]

File: lib/github.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# Return:
+#   The PR details
+#######################################
+github.pull.get () {
+    if [[ -n  ${GITHUB_TRIAGE_OAUTH:=''} ]];
+      then local OAUTH=$GITHUB_TRIAGE_OAUTH
+    elif [[ -n  ${GITHUB_READ_OAUTH:=''} ]];
+      then local OAUTH=$GITHUB_READ_OAUTH
+    else
+      local OAUTH=$GITHUB_OAUTH
```
Comment (rashidsaf) [inline lib/github.sh:27]: I think we should also validate if `GITHUB_OAUTH` exists or not

**Review by rashidsaf — APPROVED** [review]

**Review by rashidsaf — APPROVED** [review]

**Review by DevanTarunKumar — APPROVED** [review]

File: lib/github.sh (hunk 2 above)
Comment (Chekote) [reply lib/github.sh:27]: I'm going to be making a change to the way CircleCI contexts works whereby $GITHUB_OAUTH will be the only token. Doing so will simplify the checks by removing the necessity for these if statements. When I do that, I'll also add a check to ensure it exists.

**Pull request merged**